File: submissions/single-number-iii.py
# https://leetcode.com/problems/single-number-iii

from collections import Counter
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:

        # works but is too slow
        counts = Counter(nums)
        return [item for item in counts if counts[item] == 1]
        

File: submissions/single-number-iii.go
/*
 * @lc app=leetcode id=260 lang=golang
 *
 * [260] Single Number III
 */

// @lc code=start
func singleNumber(nums []int) []int {
	res := make([]int, 2)
	counts := make(map[int]int)

	for _, num := range nums {
		counts[num]++
	}

	firstAdded := false
	for k, v := range counts {
		if v == 1 {
			if firstAdded {
				res[1] = k
			} else {
				res[0] = k
				firstAdded = true
			}
		}
	}
	return res
}

// @lc code=end

